server/ipfs.js

```diff
@@ -0,0 +1,57 @@
+import ipfsAPI from 'ipfs-api';
+
+const hosts = (process.env.IPFS_HOST || 'localhost').split(',');
+const master = (process.env.IPFS_MASTER || 'localhost').split(',');
+
+const addrs = [];
+const clients = [];
+
+const initClient = async (host) => {
+  const options = {
+    host,
+    port: process.env.IPFS_PORT || '5001',
+    protocol: process.env.IPFS_PROTO || 'http',
+  };
+  const api = ipfsAPI(options);
+  const [, addr] = await api.swarm.localAddrs();
+
+  addrs.push({ addr, host });
+  clients.push({ api, host });
+};
+
+const addBootstrap = async (client) => {
+  const { api, host } = client;
+  addrs.forEach(({ addr, addrHost }) => {
+    if (addrHost !== host) {
+      api.bootstrap.add(addr);
+    }
+  });
+};
+
+(async () => {
+  await Promise.all(hosts.map(initClient));
+  if (clients.length > 1) {
+    await Promise.all(clients.map(addBootstrap));
+  }
+})();
+
+const apiClient = {
+  files: {
+    add: (stream) => {
+      const masterNodes = clients.filter(({ host }) => master.indexOf(host) > -1);
+      return Promise.race(masterNodes.map(({ api }) => api.files.add(stream)))
+    },
+    cat: (path) => Promise.race(clients.map(({ api }) => api.files.cat(path))),
```

Comment:
Since we only want to cat one file, using `race` here will yield to unnecessary IO with the IPFS.

Here we should just pick one node and cat.